src/client/client_impl.cc

```diff
@@ -158,6 +171,25 @@ google::protobuf::util::StatusOr<ReadFileChunkReply> ClientImpl::ReadFileChunk(
         "No chunk server has been found");
   }
 
+  return google::protobuf::util::Status::OK;
+}
+
+google::protobuf::util::StatusOr<ReadFileChunkReply> ClientImpl::ReadFileChunk(
+    const char* filename, size_t chunk_index, size_t offset, size_t nbytes) {
+  std::string chunk_handle;
+  uint32_t chunk_version;
+  CacheManager::ChunkServerLocationEntry chunk_server_location_entry;
+  
+  // Access the metadata first
+  auto get_metadata_status(
+      GetMetadataForChunk(filename, chunk_index, OpenFileRequest::READ,
+                          chunk_handle, chunk_version, 
+                          chunk_server_location_entry));
```

Comment:
did you forget to explicitly  pass the "force_get" parameter to the function? or is this overloaded somewhere;